cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -600,3 +617,56 @@ func isNotFound(err error) bool {
 	gerr, ok := err.(*googleapi.Error)
 	return ok && gerr.Code == http.StatusNotFound
 }
+
+func ensureNodeMatchesMetadataOrDelete(opts GCPConfig, csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest, metricsLabel string, metricsStartTime time.Time, client clientset.Interface) error {
+	// TODO: short-circuit
+
+	if !strings.HasPrefix(x509cr.Subject.CommonName, "system:node:") {
```

Comment:
This is checked by `isNodeClientCert` in `recognize`

Reply:
true, but I'm ok with belt and suspenders here